beacon-chain/core/blocks/block_operations.go

```diff
@@ -149,56 +149,29 @@ func ProcessBlockHeader(
 	beaconState *pb.BeaconState,
 	block *ethpb.BeaconBlock,
 ) (*pb.BeaconState, error) {
-	if beaconState.Slot != block.Slot {
-		return nil, fmt.Errorf("state slot: %d is different then block slot: %d", beaconState.Slot, block.Slot)
-	}
-
-	parentRoot, err := ssz.SigningRoot(beaconState.LatestBlockHeader)
+	beaconState, err := ProcessBlockHeaderNoVerify(beaconState, block)
 	if err != nil {
-		return nil, err
-	}
-	if !bytes.Equal(block.ParentRoot, parentRoot[:]) {
-		return nil, fmt.Errorf(
-			"parent root %#x does not match the latest block header signing root in state %#x",
-			block.ParentRoot, parentRoot)
+		return nil, errors.Wrap(err, "could not process block header")
 	}
 
-	bodyRoot, err := ssz.HashTreeRoot(block.Body)
-	if err != nil {
-		return nil, err
-	}
-	emptySig := make([]byte, 96)
-	beaconState.LatestBlockHeader = &ethpb.BeaconBlockHeader{
-		Slot:       block.Slot,
-		ParentRoot: block.ParentRoot,
-		StateRoot:  params.BeaconConfig().ZeroHash[:],
-		BodyRoot:   bodyRoot[:],
-		Signature:  emptySig,
-	}
-	// Verify proposer is not slashed.
 	idx, err := helpers.BeaconProposerIndex(beaconState)
```

Comment:
Any way to avoid double calling `BeaconProposerIndex` inside `ProcessBlockHeaderNoVerify` and inside `BeaconProposerIndex`? Feels a bit silly :) 
I also dont mind moving the validation of `proposer.Slashed ` to `ProcessBlockHeader`